openmetadata-ui/src/main/resources/ui/src/components/common/TestConnection/TestConnection.tsx

```diff
@@ -180,6 +185,10 @@ const TestConnection: FC<TestConnectionProps> = ({
     setMessage(TEST_CONNECTION_TESTING_MESSAGE);
     handleResetState();
 
+    // Create a new AbortController instance
+    const controller = new AbortController();
+    APICancellationControllerRef.current = controller; // Store the controller for later cancellation
```

Comment:
@Ashish8689 , can we create a custom hook for abort controller logic and use it here? this way we can reuse the logic in other components as well.


```ts

const useAbortController = () => {
  const controllerRef = useRef<AbortController | null>(null);

  useEffect(() => {
    
    const controller = new AbortController();
    controllerRef.current = controller;

    
    return () => {
      controllerRef.current?.abort();
    };
  }, []);

  return {
    controller: controllerRef.current,
  };
};

export default useAbortController;
```